internal/client/job.go

```diff
@@ -50,9 +50,18 @@ func (c *Project) job() *pb.Job {
 
 // doJob will queue and execute the job. If the client is configured for
 // local mode, this will start and target the proper runner.
-func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI) (*pb.Job_Result, error) {
+// the mon channel can be used to monitor the job status as it changes.
+// The receiver must be careful to not block sending to mon as it will block
+// the job state processing loop.
+func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI, mon chan pb.Job_State) (*pb.Job_Result, error) {
```

Comment:
Given how many things call `doJob` and don't care about `mon`, I feel like it'd be neater to do a second `doJobMonitor` function that has this signature and have `doJob` call into that with the preserved previous signature. This is a common pattern in Go libs.